pkg/controller/clone-controller.go

```diff
@@ -167,6 +167,10 @@ func (cc *CloneController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
 	pvc, err = updatePVC(cc.clientset, pvc, anno, lab)
 	if err != nil {
 		return errors.WithMessage(err, "could not update pvc %q annotation and/or label")
+	} else {
+		if phase == string(v1.PodSucceeded) {
```

Comment:
I think this is a step in the right direction.  Some style comments:

I think an `else if` would be better than separate statements.  I also think it maybe be better to change the condition to check `pvc.Annotations[AnnCloneOf] == "true"` or have a local `deletePods` bool that is set to true when `anno[AnnCloneOf] = "true"`.